### PR TITLE
MM-20526: Migrate 'components/backstage' module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by sinansonmez**
#### Summary
This PR migrates 'components/backstage' module, convert to FC (where possible) and associated tests to TypeScript

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13714
JIRA: https://mattermost.atlassian.net/browse/MM-20526

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/backstage/backstage_controller.tsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+    const scrollToTop = () => {
+        if (listRef.current) {
+            listRef.current.scrollTop = 0;
+        }
+    };
+
+    if (props.team == null || props.user == null) {
+        return <div/>;
+    }
```
Comment (hmhealey) [inline components/backstage/backstage_controller.tsx:94]: Slight changes here since I think this is just written this way since it's just old code. There's nothing particularly wrong, but it's an old style

```suggestion
    if (!props.team || !props.user) {
        return null;
    }
```

File: components/backstage/backstage_controller.tsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    enableCommands: boolean;
+    enableOAuthServiceProvider: boolean;
+    canCreateOrDeleteCustomEmoji: boolean;
+    canManageIntegrations: boolean;
+}
+
+const BackstageController = (props: Props) => {
+    const listRef = useRef<HTMLDivElement>(null);
+
+    const scrollToTop = () => {
```
Comment (hmhealey) [inline components/backstage/backstage_controller.tsx:86]: This might need to be wrapped in a `useCallback` since it's passed through to the page components

**Review by hmhealey — CHANGES_REQUESTED** [review]

File: components/backstage/backstage_controller.tsx (hunk 2 above)
Comment (sinansonmez) [reply components/backstage/backstage_controller.tsx:86]: I also thought that but every time when I rep `ref` inside `useCallback` I am encountering a strange behavior. I did a quick research and this maybe the reason of the strange behavior: https://stackoverflow.com/questions/61172383/getting-ref-null-inside-usecallback-hook

TLDR: `useRef` doesn't notify you when its content changes, which doesn't trigger the update of `useCallback` function.

Comment (hmhealey) [reply components/backstage/backstage_controller.tsx:86]: Huh, I thought the value of `listRef` would be consistent between renders, and it was just `listRef.current` that might change. Good find

**Review by AshishDhama — CHANGES_REQUESTED** [review]

**Review by AshishDhama — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @sinansonmez 
Tested, looks good to merge.
[E2E report ](https://mattermost-cypress-report.s3.amazonaws.com/246935-f210925-onprem-ent-webapp-pr-11120-mm-ee-test:f210925/mochawesome.html)- no PR related failures.
